File: design_patterns_101/Creational Patterns/Singleton Pattern/singletonPattern.js
class Singleton {
    constructor() {
      if (!Singleton.instance) {
        Singleton.instance = this;
      }
      return Singleton.instance;
    }
  
    someBusinessLogic() {
      console.log("Singleton instance is doing something...");
    }
  }
  
  // Usage
  const instance1 = new Singleton();
  const instance2 = new Singleton();
  
  console.log(instance1 === instance2); // Output: true
  instance1.someBusinessLogic();
  

File: design_patterns_101/Creational Patterns/Singleton Pattern/singletonPattern.py
class Singleton:
    _instance = None
    
    @staticmethod
    def get_instance():
        if Singleton._instance is None:
            Singleton._instance = Singleton()
        return Singleton._instance
    
    def __init__(self):
        if Singleton._instance is not None:
            raise Exception("This class is a Singleton!")
    
    def some_business_logic(self):
        print("Singleton instance is doing something...")
        

# Usage
singleton_instance = Singleton.get_instance()
singleton_instance.some_business_logic()
